### PR TITLE
Add job for running queries

### PR DESCRIPTION
**Pull request by tseaver**
~~Uses #1068 as a base.~~

### CODE REVIEW

**Comment by tseaver** [comment]
@dhermes Rebased after merge of #1068.  PTAL.

File: gcloud/bigquery/client.py [hunk 1]
```diff
@@ -153,3 +154,17 @@ def extract_table_to_storage(self, name, source, *destination_uris):
         """
         return ExtractTableToStorageJob(name, source, destination_uris,
                                         client=self)
+
+    def run_query(self, name, query):
+        """Construct a job for running a BG SQL query.
```
Comment (dhermes) [inline gcloud/bigquery/client.py]: What is a `BG SQL` query?

File: gcloud/bigquery/job.py [hunk 2]
```diff
@@ -1140,3 +1153,307 @@ def _build_resource(self):
...
+    _default_dataset = None
+    _destination_table = None
+    _flatten_results = None
+    _priority = None
+    _use_query_cache = None
+    _write_disposition = None
+
+
+class RunQueryJob(_BaseJob):
+    """Asynchronous job: query BQ tables.
```
Comment (dhermes) [inline gcloud/bigquery/job.py]: This makes me think the `BG` above is just meant to be `BQ`. Please just write out the whole word / phrase.

**Comment by dhermes** [comment]
1. System test forthcoming?
1. Similarly, have you compared our BigQuery system tests to those in `gcloud-node` (I keep forgetting to bring this up, sorry if it's out of place here)
1. This all seems very vanilla / identical to existing. Is it worth have something more generic we can use (included generated tests for getters, setters and deleters, or just a generic subclass of `property` that we can test once)?
1. How close to "done" with BigQuery? (Cut release at that point?)

File: gcloud/bigquery/client.py (hunk 1 above)
Comment (tseaver) [inline gcloud/bigquery/client.py]: 73f654b

**Comment by dhermes** [comment]
Thanks for the tweaks. As for my other questions?

**Comment by tseaver** [comment]
> System test forthcoming?

Yes, shortly.

> Similarly, have you compared our BigQuery system tests to those in gcloud-node (I keep forgetting to bring this up, sorry if it's out of place here).

I've looked at them:  we have similar coverage, except one missing for #1068.

> This all seems very vanilla / identical to existing. Is it worth have something more generic we can use (included generated tests for getters, setters and deleters, or just a generic subclass of property that we can test once)?

We could bag the property/validation stuff, and just rely on docs for the user-settable configuration attributes.  Or we could write a 'validating descriptor' helper and use that instead of custom property methods.

> How close to "done" with BigQuery? (Cut release at that point?)

We don't have coverage for the `/upload` APIs (multipart/resumable, just like storage).  And we're missing `Client.list_jobs`.  Other than that, I think we're close.

**Comment by tseaver** [comment]
Rebased on top of #1075.

**Comment by dhermes** [comment]
Ping me when #1075 is in.

**Comment by tseaver** [comment]
@dhermes rebased after merge of #1075.

**Comment by dhermes** [comment]
LGTM

**Pull request merged**